app/models/variant.rb

```diff
@@ -0,0 +1,24 @@
+class Variant < ApplicationRecord
+
+  belongs_to :assessment
+  belongs_to :format
+  has_many :solutions, inverse_of: :variant, dependent: :destroy
+  has_many :assets, as: :owner, dependent: :destroy
+
+  validates :content, presence: true
+  validates :assessment, presence: true
+  validates :source_identifier, uniqueness: { scope: :assessment }, allow_blank: true
+
+  before_destroy :abort_unless_destroyed_by_association
+
+  def other_variants
+    assessment ? assessment.variants.without(self) : []
+  end
+
+  protected
+
+  def abort_unless_destroyed_by_association
+    throw(:abort) unless destroyed_by_association
+  end
+
+end
```

Comment:
Am I missing the `attr_accessor` declaration for `destroyed_by_association` which is checked by `abort_unless_destroyed_by_association `?

Reply:
It is actually a built-in Rails method - I was not previously aware of it, but it is set if a record is being destroyed because its parent is being destroyed; and it is `nil` otherwise.

Reply:
oh nifty!